module/vision/GoalDetector/src/GoalDetector.cpp

```diff
@@ -262,105 +292,6 @@ namespace module::vision {
                         }
                     }
 
-                    log<NUClear::DEBUG>("Culling colinear goals");
-
-                    for (auto it1 = goals->goals.begin(); it1 != goals->goals.end(); it1 = std::next(it1)) {
-                        log<NUClear::DEBUG>(fmt::format("Post Disances {}", it1->post.distance));
-                    }
-                    // If a sided goal is "behind" another sided goal it cannot be a goal
-                    // loop through all goals (sided), if any other goal is behind it (ie ~same phi theta && goal1_dist
-                    // < goal2_dist) throw it away! (or for now, make it an unknow goal or something)
-                    const Eigen::Affine3f Hcw(horizon.Hcw.cast<float>());
-                    for (auto goal_post_a = goals->goals.begin(); goal_post_a != goals->goals.end();) {
-                        bool increment_a = false;
-                        // Vector from camera to the bottom center of the post, projected to the observation
-                        Eigen::Matrix<float, 2, 1> rGOo_a =
-                            (goal_post_a->post.bottom.head<2>() * goal_post_a->post.distance);
-
-                        // Length of that vector
-                        const float distance_a = rGOo_a.norm();
-
-                        rGOo_a /= distance_a;
-
-                        for (auto goal_post_b = goals->goals.begin(); goal_post_b != goals->goals.end();) {
```

Comment:
Was all this supposed to be removed?

Reply:
Yeah, we changed the metric and decided merging clusters would be better than a colinear solution

Reply:
got it